web-admin/src/features/projects/github/GithubRepoUpdater.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  adminServiceListGithubUserRepos,
+  createAdminServiceGetGithubUserStatus,
+  createAdminServiceListGithubUserRepos,
+  getAdminServiceListGithubUserReposQueryKey,
+  type RpcStatus,
+} from "@rilldata/web-admin/client";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+import { waitUntil } from "@rilldata/web-common/lib/waitUtils";
+import { derived, get } from "svelte/store";
+
+export class GithubRepoUpdater {
+  public readonly userStatus = createAdminServiceGetGithubUserStatus();
+  public readonly userRepos = derived([this.userStatus], ([userStatus], set) =>
+    createAdminServiceListGithubUserRepos({
+      query: {
+        enabled: !!userStatus.data?.hasAccess,
+        queryClient,
+      },
+    }).subscribe(set),
+  ) as ReturnType<
+    typeof createAdminServiceListGithubUserRepos<
+      Awaited<ReturnType<typeof adminServiceListGithubUserRepos>>,
+      RpcStatus
+    >
+  >;
+
+  public readonly status = derived(
+    [this.userStatus, this.userRepos],
+    ([userStatus, userRepos]) => {
+      if (userStatus.isFetching || userRepos.isFetching) {
+        return {
+          isLoading: true,
+          error: undefined,
+        };
+      }
```

Comment:
We should be consistent with terminology. If reading `isFetching`, then we should return `isFetching`. If reading `isLoading`, then we should return `isLoading`.